### ISSUE TITLE
WebGPU: Tests asserting in tests/compute

### ISSUE THREAD

**Issue by aleino-nv** [shader-slang/slang#5172]
The following tests are currently asserting for `wgpu` and are therefore disabled for this API.
This task is about triaging these issues -- either fix on the spot or file a more detailed task.

1. tests/compute/atomics-buffer.slang
2. tests/compute/atomics-groupshared.slang
3. tests/compute/atomics.slang
4. tests/compute/buffer-type-splitting.slang
5. tests/compute/byte-address-buffer.slang
6. tests/compute/column-major.slang
7. tests/compute/constant-buffer-memory-packing.slang
8. tests/compute/func-cbuffer-param.slang
9. tests/compute/parameter-block.slang
10. tests/compute/texture-get-dimensions.slang
11. tests/compute/texture-sampling.slang
12. tests/compute/texture-simple.slang
13. tests/compute/texture-simpler.slang
14. tests/compute/transcendental-double.slang

**Comment by aleino-nv** [comment]
It turns out that the asserts are a bit spurious in the sense that they just mean "something went wrong" rather than being checks that the internal logic of slang-test is ok.

These failures boil down to these 6 categories:

1. **Capabilities: using of 'InterlockedAdd'**
   - Tests:
      1. atomics-buffer
      2. atomics-groupshared
      3. atomics
   - Example output:
````
tests/compute/atomics.slang(23): error 36107: entrypoint 'computeMain' does not support compilation target 'wgsl' with stage 'compute'
tests/compute/atomics.slang(17): note: see using of 'InterlockedAdd'
````
2. **Capabilities: using of 'Load'**
   - Tests:
      1. buffer-type-splitting
      2. byte-address-buffer
   - Example output:
````
tests/compute/byte-address-buffer.slang(35): error 36107: entrypoint 'computeMain' does not support compilation target 'wgsl' with stage 'compute'
tests/compute/byte-address-buffer.slang(23): note: see using of 'Load'
````
3. **WebGPU error: Binding doesn't exist**
   - Tests:
      1. column-major
      2. constant-buffer-memory-packing
      3. func-cbuffer-param
      4. parameter-block
4. **WebGPU error: unresolved type 'texture_1d_array'**
   - Tests:
      1. texture-get-dimensions
      2. texture-sampling
      3. texture-simple
   - Example output:
````
WGPU error: Error while parsing WGSL: :11:40 error: unresolved type 'texture_1d_array'
````
5. **Slang: unknown system-value semantic 'sv_groupindex'**
   - Tests:
      1. texture-simpler
   - Example output:
````
tests/compute/texture-simpler.slang(18): error 40006: unknown system-value semantic 'sv_groupindex'
````
6. **Slang: unexpected: 'double' type emitted (WGSL emitter)**
   - Tests:
      1. transcendental-double
   - Example output:
````
unexpected: 'double' type emitted
````

**Comment by aleino-nv** [comment]
I think 5 and 6 already have known issues. (I should add links to those issues where I disable the tests though.)
I will file issues to track any remaining issues that don't already have known related issues.

**Comment by aleino-nv** [comment]
I've split this issue into a few different tasks, after some initial investigation:

1. Split into new issue #5220
2. Split into new issue #5221
3. Split into new issue #5222
4. Split into new issue #5223
5. Will be addressed by existing task #4940
6. Will be addressed by existing task #4947
